core/functional_unroller.h

```diff
@@ -0,0 +1,89 @@
+/*********************                                                        */
+/*! \file functional_unroller.h
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann
+** This file is part of the pono project.
+** Copyright (c) 2019 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief An unroller implementation for functional transition systems that has
+**        a configurable parameter for when to introduce new timed variables.
+**
+**
+**/
+#pragma once
+
+#include "core/unroller.h"
+
+namespace pono {
+class FunctionalUnroller : public Unroller
+{
+ public:
+  /** Instantiate a FunctionalUnroller
+   *  @param ts the transition system to unroll
+   *  @param solver the solver to use
+   *  @param interval -- the interval to introduce new timed variables for state
```

Comment:
Hmm ran `clang-format` on this file and it didn't change anything. You're thinking for line length?

Reply:
i was not able to completely understand the text. I guess, you wanted some line breaks in them. That is why I asked

Reply:
Oh I see! Yeah I think that was because of `clang-format` actually. Fixed in https://github.com/upscale-project/pono/pull/135/commits/35ff261eba8eeab6fdc2317b2f650dec84712843